src/server/service/kv.rs

```diff
@@ -177,9 +183,12 @@ macro_rules! handle_request {
             let begin_instant = Instant::now();
 
             let source = req.mut_context().take_request_source();
-            let resource_group_name = req.get_context().get_resource_group_name();
+            let resource_control_ctx = req.get_context().get_resource_control_context();
```

Comment:
Please move this code into a function, it repeats several times.

Reply:
can't because the reqs are of differrent type

Reply:
But you can just pass resource_control_ctx as the parameter